contracts/5/SushiswapPool12Goblin.sol

```diff
@@ -0,0 +1,243 @@
+pragma solidity 0.5.16;
+import "openzeppelin-solidity-2.3.0/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity-2.3.0/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity-2.3.0/contracts/utils/ReentrancyGuard.sol";
+import "openzeppelin-solidity-2.3.0/contracts/token/ERC20/IERC20.sol";
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol";
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Pair.sol";
+import "@uniswap/v2-core/contracts/libraries/Math.sol";
+import "./uniswap/IUniswapV2Router02.sol";
+import "./Strategy.sol";
+import "./SafeToken.sol";
+import "./Goblin.sol";
+import "./interfaces/IMasterChef.sol";
+
+// SushiswapPool12Goblin is specific for SUSHI-ETH pool in Sushiswap.
+// In this case, fToken = SUSHI and pid = 12.
+contract SushiswapPool12Goblin is Ownable, ReentrancyGuard, Goblin {
+    /// @notice Libraries
+    using SafeToken for address;
+    using SafeMath for uint256;
+
+    /// @notice Events
+    event Reinvest(address indexed caller, uint256 reward, uint256 bounty);
+    event AddShare(uint256 indexed id, uint256 share);
+    event RemoveShare(uint256 indexed id, uint256 share);
+    event Liquidate(uint256 indexed id, uint256 wad);
+
+    /// @notice Immutable variables
+    IMasterChef public masterChef;
+    IUniswapV2Factory public factory;
+    IUniswapV2Router02 public router;
+    IUniswapV2Pair public lpToken;
+    address public weth;    
+    address public sushi;
+    address public operator;
+    uint256 public pid;
+
+    /// @notice Mutable state variables
+    mapping(uint256 => uint256) public shares;
+    mapping(address => bool) public okStrats;
+    uint256 public totalShare;
+    Strategy public addStrat;
+    Strategy public liqStrat;
+    uint256 public reinvestBountyBps;
+
+    constructor(
+        address _operator,
+        IMasterChef _masterChef,
+        IUniswapV2Router02 _router,        
+        Strategy _addStrat,
+        Strategy _liqStrat,
+        uint256 _reinvestBountyBps
+    ) public {
+        operator = _operator;
+        weth = _router.WETH();
+        masterChef = _masterChef;
+        router = _router;
+        factory = IUniswapV2Factory(_router.factory());        
+        pid = 12;
```

Comment:
let's make this a constant at the top. `uint256 public constant pid = 12`

Reply:
done